rubin_sim/maf/mafContrib/filterPairTGapsMetric.py

```diff
@@ -0,0 +1,175 @@
+                
+                dT = np.concatenate(dT) if len(dT)>0 else np.array(dT)
+        else:
+            # collect only nearest 
+            if f0==f1:
+                # get diagonal ones nearest nonzero, offset=1
+                #dT = np.diagonal(diffmat, offset=1)
+                dT = np.diff(timeCol0)
+            else:
+                timeCol0 = dataSlice['observationStartMJD'][idx0]
+                timeCol1 = dataSlice['observationStartMJD'][idx1]
+
+                timeCol0 = timeCol0.reshape((len(timeCol0), 1))
+                timeCol1 = timeCol1.reshape((len(timeCol1), 1))
+
+                # calculate time gaps matrix
+                diffmat = np.subtract(timeCol0, timeCol1.T)
+                # get tgaps both left and right
+                # keep only negative ones
+                masked_ar = np.ma.masked_where(diffmat>=0, diffmat, )
+                left_ar = np.max(masked_ar, axis=1)
+                dT_left = -left_ar.data[~left_ar.mask]
+                
+                # keep only positive ones
+                masked_ar = np.ma.masked_where(diffmat<=0, diffmat)
+                right_ar = np.min(masked_ar, axis=1)
+                dT_right = right_ar.data[~right_ar.mask]
+                
+                dT = np.concatenate([dT_left.flatten(), dT_right.flatten()])
+        
+        return dT
+    
+    def run(self, dataSlice, slicePoint=None):
+                        
+        # sort the dataSlice in order of time.  
+        dataSlice.sort(order='observationStartMJD')
+        
+        fieldRA = np.mean(dataSlice['fieldRA']) 
+        fieldDec = np.mean(dataSlice['fieldDec'])
+        pixId = radec2pix(nside=16, ra=np.radians(fieldRA), dec=np.radians(fieldDec))
+        
+        fom_dic = {}
+        dT_dic = {}
+        for fltpair in self.fltpairs:
+            dT = self._get_dT(dataSlice, fltpair[0], fltpair[1])
+            # calculate FoM
+            # cut out at average
+            # std below 
+            # FOM = Nv  / std
+            
+            if fltpair[0]==fltpair[1]:
+                bins = self.bins_same
+            else:
+                bins = self.bins_diff
+                dT_dic[fltpair] = dT
+                
+            hist, _ = np.histogram(dT, bins=bins)
+            
+            if np.any(hist):
+                hist_clip = np.clip(hist, a_min=0, a_max=self.Nv_clip[fltpair])
+                fom = hist.sum() / hist_clip.std()
+            else:
+                fom = np.nan
+            fom_dic[fltpair] = fom
+            
+            print(pixId, fltpair, fom)
+            
+        if self.save_dT:
+            result = {
+                'pixId': pixId,
+                'dT_dic': dT_dic,
+                'fom_dic': fom_dic,
+                'dataSlice': dataSlice
+                  }
+        else:
+            result = {
+                'pixId': pixId,
+                'fom_dic': fom_dic,
+                  }
+        return result
```

Comment:
Okay, so I thought when we talked, we had discussed this metric actually providing a single value per healpix point? It looks like you are gearing up here to return the clipped rms histogram value per filter pair. 

I would gently suggest: 
If you're feeling like you've debugged this metric, drop the first "if self.save_dT" option. (if you need to debug in the future, you can always modify the metric in place and just return all of these values, then go back to "normal". You probably do not need to return the dataSlice values, as you can recreate those from the slicer). 

You do not need to return the pixId, it is preserve in other places in the metricValues and slicer combination.

Possibly consider only returning the contents of the 'reduce_tgaps_FoM' in place of the fom_dic, if you are not planning on running this on your own (and saving the output) .. we won't be saving the full outputs and you will not have this information available, so it just ends up increasing the memory footprint of the metric unnecessarily.